### DISCUSSION TITLE
ScriptReference/RenderSettings

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#509] [Notes]
# ScriptReference/RenderSettings

https://docs.unity3d.com/ScriptReference/RenderSettings.html

**Comment by HeliosDoubleSix** [comment]
To change sky gradient colors in HDRP:
https://forum.unity.com/threads/change-sky-gradient-in-a-script-hdrp.812481/
using UnityEngine.Rendering;

`Volume volume = GetComponent<Volume>();
GradientSky sky;
volume.profile.TryGet(out sky);
sky.top.value = new Color(0, 0, 1);`

-

Skybox is a material so you can set values on that via:
`RenderSettings.skybox.SetFloat ( "_Exposure", yourExposure );`

-

To in editor only make and store a copy of the entire settings object ( as JSON ) to move it to another scene or restore it later:

https://assetstore.unity.com/packages/tools/particles-effects/render-settings-duplicator-58831

```
private static Object GetRenderSettings()
{
    return typeof(RenderSettings).GetMethod("GetRenderSettings", BindingFlags.Static | BindingFlags.NonPublic).Invoke(null, null) as Object;
}
 
`private static Object GetLightmapSettings()
{
    return typeof(LightmapEditorSettings).GetMethod("GetLightmapSettings", BindingFlags.Static | BindingFlags.NonPublic).Invoke(null, null) as Object;
}
 
RenderSettings = GameObject.Instantiate(GetRenderSettings());
 
renderSettingsJSON = UnityEditor.EditorJsonUtility.ToJson(RenderSettings);
lightmapSettingsJSON = UnityEditor.EditorJsonUtility.ToJson(LightmapSettings);
 
EditorJsonUtility.FromJsonOverwrite(snap.renderSettingsJSON, GetRenderSettings());
EditorJsonUtility.FromJsonOverwrite(snap.lightmapSettingsJSON, GetLightmapSettings());
 
foreach (EditorWindow window in Resources.FindObjectsOfTypeAll<EditorWindow>cry)) window.Repaint();

EditorSceneManager.MarkSceneDirty(SceneManager.GetActiveScene());
```



-

See also

UnityEditor.Unsupported.GetRenderSettings();
https://github.com/Unity-Technologies/UnityCsReference/blob/master/Editor/Mono/Unsupported.bindings.cs

LightingEditor.cs source:
https://github.com/Unity-Technologies/UnityCsReference/blob/70832e6c5b834ed12c9c19d91d6b710c55dfea67/Editor/Mono/SettingsWindow/LightingEditor.cs
